examples/gno.land/r/gnoland/boards2/v1/permissions_default.gno

```diff
@@ -0,0 +1,273 @@
+package boards2
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/nt/commondao"
+
+	"gno.land/r/sys/users"
+)
+
+// DefaultPermissions manages users, roles and permissions.
+type DefaultPermissions struct {
+	superRole Role
+	dao       *commondao.CommonDAO
+	users     *avl.Tree // string(std.Address) -> []Role
+	roles     *avl.Tree // string(role) -> []Permission
+}
+
+// NewDefaultPermissions creates a new permissions type.
+// This type is a default implementation to handle users, roles and permissions.
+func NewDefaultPermissions(dao *commondao.CommonDAO) *DefaultPermissions {
+	if dao == nil {
+		panic("default permissions require a DAO")
+	}
+
+	dp := &DefaultPermissions{
+		dao:   dao,
+		roles: avl.NewTree(),
+		users: avl.NewTree(),
+	}
+
+	// Make sure that DAO users are indexed when DAO has members
+	dao.Members().IterateByOffset(0, dao.Members().Size(), func(m std.Address) bool {
+		dp.users.Set(m.String(), []Role(nil))
+		return false
+	})
+
+	return dp
+}
+
+// SetSuperRole assigns a super role.
+// A super role is one that have all permissions.
+// These type of role doesn't need to be mapped to any permission.
+func (dp *DefaultPermissions) SetSuperRole(r Role) {
+	dp.superRole = r
+}
+
+// AddRole add a role with one or more assigned permissions.
+func (dp *DefaultPermissions) AddRole(r Role, p Permission, extra ...Permission) {
+	dp.roles.Set(string(r), append([]Permission{p}, extra...))
+}
```

Comment:
```suggestion
func (dp *DefaultPermissions) AddRole(r Role, perms ...Permission) {
	dp.roles.Set(string(r), perms)
}
```

Reply:
The idea here was to require that when a role is added it would at least have a permission assigned, optionally more, without the need to check `len(perms) > 0` to the return an error or panic, to keep it simple.

I would prefer to keep it simple, unless there is a reason for a role without permissions 🤔

Currently a role without permissions won't be able to do anything.